zeroex/order_validator.go

```diff
@@ -197,6 +207,155 @@ func (o *OrderValidator) BatchValidate(signedOrders []*SignedOrder) map[common.H
 	return orderHashToInfo
 }
 
+// BatchOffchainValidation performs all off-chain validation checks on a batch of 0x orders.
+// These checks include:
+// - `MakerAssetAmount` and `TakerAssetAmount` cannot be 0
+// - `AssetData` fields contain properly encoded, and currently supported assetData (ERC20 & ERC721 for now)
+// - `Signature` contains a properly encoded 0x signature
+// - Validate that order isn't expired
+// Returns an orderHashToInfo mapping with all invalid orders added to it, and an array of the valid signedOrders
+func (o *OrderValidator) BatchOffchainValidation(signedOrders []*SignedOrder) (map[common.Hash]*OrderInfo, []*SignedOrder) {
```

Comment:
When we change the API for `BatchValidate`, we should take another look at this method as well. I'm okay with leaving it as is for now.